pkg/worker/image.go

```diff
@@ -256,6 +256,75 @@ func (c *ImageClient) InspectAndVerifyImage(ctx context.Context, sourceImage str
 	return nil
 }
 
+func (c *ImageClient) BuildAndArchiveImage(ctx context.Context, dockerfile string, imageId string, buildCtxPath string) error {
+	buildPath, err := os.MkdirTemp("", "")
+	if err != nil {
+		return errors.Wrap(err, "create temp dir")
+	}
+	defer os.RemoveAll(buildPath)
+	tempDockerFile := filepath.Join(buildPath, "Dockerfile")
+	f, err := os.Create(tempDockerFile)
+	if err != nil {
+		return errors.Wrap(err, "create temp dockerfile")
+	}
+	fmt.Fprintf(f, dockerfile)
+	f.Close()
+
+	imagePath := filepath.Join(buildPath, "image")
+	ociPath := filepath.Join(buildPath, "oci")
+	tempBundlePath := filepath.Join(c.imageBundlePath, imageId)
+	os.MkdirAll(imagePath, 0755)
+	os.MkdirAll(ociPath, 0755)
+
+	cmd := exec.Command("buildah", "--root", imagePath, "bud", "-f", tempDockerFile, "-t", imageId+":latest", buildCtxPath)
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		return errors.Wrap(err, "buildah bud")
+	}
+
+	cmd = exec.Command("buildah", "--root", imagePath, "push", imageId+":latest", "oci:"+ociPath+":latest")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
+	err = cmd.Run()
+	if err != nil {
+		return errors.Wrap(err, "buildah push")
+	}
+
+	engine, err := dir.Open(ociPath)
+	if err != nil {
+		return errors.Wrap(err, "open CAS "+ociPath)
+	}
+	defer engine.Close()
+
+	unpackOptions := umociUnpackOptions()
+
+	engineExt := casext.NewEngine(engine)
+	defer engineExt.Close()
+
+	err = umoci.Unpack(engineExt, "latest", tempBundlePath, unpackOptions)
+	if err != nil {
+		return errors.Wrap(err, "unpack image")
+	}
+
+	for _, dir := range requiredContainerDirectories {
+		fullPath := filepath.Join(tempBundlePath, "rootfs", dir)
+		err := os.MkdirAll(fullPath, 0755)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("creating /%s directory", dir))
+		}
+	}
+
+	defer os.RemoveAll(tempBundlePath)
```

Comment:
Why is this defer here and not right after we create tempBundlePath?

Reply:
Moved it up